agreement/keyManager_test.go

```diff
@@ -53,11 +56,18 @@ func (m *recordingKeyManager) DeleteOldKeys(r basics.Round) {
 }
 
 // Record implements KeyManager.Record.
-func (m *recordingKeyManager) RecordAsync(acct basics.Address, round basics.Round, action account.ParticipationAction) {
+func (m *recordingKeyManager) Record(acct basics.Address, round basics.Round, action account.ParticipationAction) {
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
 	if _, ok := m.recording[acct]; !ok {
 		m.recording[acct] = make(map[account.ParticipationAction]basics.Round)
 	}
 	m.recording[acct][action] = round
 }
+
+func (m *recordingKeyManager) ValidateVoteRound(t *testing.T, address basics.Address, round basics.Round) {
+	m.mutex.Lock()
+	assert.Equal(t, round, m.recording[address][account.Vote])
+	assert.Equal(t, round, m.recording[address][account.BlockProposal])
```

Comment:
fyi : be aware that usually using `require` is better than `assert` : `assert` would trigger a failure once the test is over, while `require` would fail right away. One important place where `require` is better is that we can test expressions that would cause the test to fail due to the wrong reason later on. ( i.e. testing for nil value in return )